packages/core/lib/Configuration.ts

```diff
@@ -35,7 +36,7 @@ export let CONFIG: Readonly<RemoveIndex<ArgumentsObject>>;
 export class Configuration {
   initialize(argumentValues: ArgumentsObject) {
     if (CONFIG) {
-      throw Error("Already initialized the config singleton!");
+      throw new Error(singleTonAlreadySet("config"));
```

Comment:
In most languages, we would extend the error with our own type. Is this a JS convention? This is definitely better than what we had before, but I wonder if extending the Error class would not be better. In that case, we can also check what type of error is thrown if we want to catch it.

Additionally, we should maybe look into adding a `cause` like mentioned [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error)

Reply:
Yes we could do that. I don't really find a strict convention on this on the internet. I copied this setup from the typescript source code. Checking the type would indeed be the only benefit but it would look like this:

```
try {
  ...
} catch (e) {
  if (e instanceof MyCustomError) {
    ...
  } else {
    ...
  }
}
```

while with the current setup you can also do this:
```
try {
  ...
} catch (e) {
  if (e.message.includes('MyCustomError')) {
    ...
  } else {
    ...
  }
}
```

The only difference between Error and MyCustomError would be the name.


The cause property is not supported by all node versions. Only 16.9.0 and higher.
[It is used when catching and re-throwing an error with a more-specific or useful error message in order to still have access to the original error.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause)
So I do not think it is really relevant here.

Reply:
Okay, since this is mostly syntactic sugar anyway for JavaScript, let's stick with the current proposed method